node/src/container_chain_spawner.rs

```diff
@@ -312,10 +405,13 @@ impl ContainerChainSpawner {
                             Err(e) => panic!("{} failed: {}", name, e),
                         }
                     }
-                    _ = on_exit_future => {
-                        // Graceful shutdown
+                    stop_unassigned = on_exit_future => {
+                        // Graceful shutdown.
+                        // `stop_unassigned` will be `Ok` if `.stop()` has been called, which means that the
+                        // container chain has been unassigned, and will be `Err` if the handle has been dropped,
+                        // which means that the node is stopping.
                         // Delete existing database if running as collator
-                        if validator {
+                        if validator && stop_unassigned.is_ok() && !container_chain_cli.base.keep_db && warp_sync {
```

Comment:
```suggestion
                        if validator && stop_unassigned.is_ok() && !container_chain_cli.base.keep_db {
```

When stopping a container chain we want to always delete the db to save space.